isisd/isis_tlvs.c

```diff
@@ -1898,6 +2580,211 @@ static int unpack_item_ipv6_reach(uint16_t mtid, uint8_t len, struct stream *s,
 	return 1;
 }
 
+/* Functions related to TLV 242 Router Capability */
+static struct isis_router_cap *copy_tlv_router_cap(
+			       const struct isis_router_cap *router_cap)
+{
+	struct isis_router_cap *rv = XMALLOC(MTYPE_ISIS_TLV, sizeof(*rv));
+
+	if (!router_cap)
+		return NULL;
+
+	memcpy(rv, router_cap, sizeof(*rv));
+
+	return rv;
+}
+
+static void format_tlv_router_cap(const struct isis_router_cap *router_cap,
+				  struct sbuf *buf, int indent)
+{
+	char addrbuf[INET_ADDRSTRLEN];
+
+	if (!router_cap)
+		return;
+
+	/* Router ID and Flags */
+	inet_ntop(AF_INET, &router_cap->router_id, addrbuf, sizeof(addrbuf));
+	sbuf_push(buf, indent, "Router Capability:");
+	sbuf_push(buf, indent, " %s , D:%c, S:%c\n", addrbuf,
+		  router_cap->flags & ISIS_ROUTER_CAP_FLAG_D ? '1' : '0',
+		  router_cap->flags & ISIS_ROUTER_CAP_FLAG_S ? '1' : '0');
+
+	/* SR Global Block */
+	if (router_cap->srgb.range_size != 0)
+		sbuf_push(buf, indent,
+			"  Segment Routing: I:%s V:%s, SRGB Base: %d Range: %d\n",
+			IS_SR_IPV4(router_cap->srgb) ? "1" : "0",
+			IS_SR_IPV6(router_cap->srgb) ? "1" : "0",
+			router_cap->srgb.lower_bound,
+			router_cap->srgb.range_size);
+
+	/* SR Algorithms */
+	if (router_cap->algo[0] != SR_ALGORITHM_UNSET) {
+		sbuf_push(buf, indent, "    Algorithm: %s",
+			  router_cap->algo[0] == 0 ? "0: SPF"
+						   : "0: Strict SPF");
+		for (int i = 0; i < SR_ALGORITHM_COUNT; i++)
```

Comment:
Shouldn't we start from `1` here instead since `router_cap->algo[0]` is displayed separately above?

Reply:
Yes. For sure